Embulk return exit code 0 when error occurs.(want help, is this a bug?)

I am using the newest version 0.8.30 of embulk and when I make some error in yml file, I found that embulk will show error messages but the exit code is still 0.
I tested versions 0.8.26 and 0.8.27, and found that this occurred from version 0.8.28.
Below are my commands and error messages, please check and help me. I really need it to return an error code when embulk fails.

`root@6cd9bedd9b83:~# embulk run test.yml
2017-08-28 02:47:41.605 +0000: Embulk v0.8.30
2017-08-28 02:47:50.414 +0000 [INFO] (0001:transaction): Loaded plugin embulk-input-mysql (0.8.5)
2017-08-28 02:47:58.547 +0000 [INFO] (0001:transaction): Loaded plugin embulk-output-bigquery (0.4.5)
2017-08-28 02:47:58.624 +0000 [INFO] (0001:transaction): JDBC Driver = /root/.embulk/jruby/2.3.0/gems/embulk-input-mysql-0.8.5/default_jdbc_driver/mysql-connector-java-5.1.34.jar
2017-08-28 02:47:58.637 +0000 [INFO] (0001:transaction): Fetch size is 10000. Using server-side prepared statement.
2017-08-28 02:47:59.075 +0000 [INFO] (0001:transaction): Using JDBC Driver mysql-connector-java-5.1.34 ( Revision: jess.balint@oracle.com-20141014163213-wqbwpf1ok2kvo1om )
2017-08-28 02:47:59.075 +0000 [WARN] (0001:transaction): This plugin will update MySQL Connector/J version in the near future release.
2017-08-28 02:47:59.076 +0000 [WARN] (0001:transaction): It has some incompatibility changes.
2017-08-28 02:47:59.076 +0000 [WARN] (0001:transaction): For example, the 5.1.35 introduced `noTimezoneConversionForDateType` and `cacheDefaultTimezone` options.
2017-08-28 02:47:59.076 +0000 [WARN] (0001:transaction): Please read a document and make sure configuration carefully before updating the plugin.
2017-08-28 02:47:59.080 +0000 [WARN] (0001:transaction): The client timezone(Etc/UTC) is different from the server timezone(GMT+07:00). The plugin will fetch wrong datetime values.
2017-08-28 02:47:59.081 +0000 [WARN] (0001:transaction): You may need to set options `useLegacyDatetimeCode` and `serverTimezone`
2017-08-28 02:47:59.081 +0000 [WARN] (0001:transaction): Example. `options: { useLegacyDatetimeCode: false, serverTimezone: UTC }`
2017-08-28 02:47:59.083 +0000 [WARN] (0001:transaction): The plugin will set `useLegacyDatetimeCode=false` by default in future.
org.embulk.exec.PartialExecutionException: java.lang.RuntimeException: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'test.test_data_table' doesn't exist
        at org.embulk.exec.BulkLoader$LoaderState.buildPartialExecuteException(BulkLoader.java:375)
        at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:607)
        at org.embulk.exec.BulkLoader.access$000(BulkLoader.java:35)
        at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:391)
        at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:387)
        at org.embulk.spi.Exec.doWith(Exec.java:25)
        at org.embulk.exec.BulkLoader.run(BulkLoader.java:387)
        at org.embulk.EmbulkEmbed.run(EmbulkEmbed.java:180)
        at org.embulk.EmbulkRunner.runInternal(EmbulkRunner.java:350)
        at org.embulk.EmbulkRunner.run(EmbulkRunner.java:184)
        at org.embulk.cli.EmbulkRun.runSubcommand(EmbulkRun.java:473)
        at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:99)
        at org.embulk.cli.Main.main(Main.java:28)
        Suppressed: java.lang.NullPointerException
                at org.embulk.exec.BulkLoader.doCleanup(BulkLoader.java:496)
                at org.embulk.exec.BulkLoader$3.run(BulkLoader.java:427)
                at org.embulk.exec.BulkLoader$3.run(BulkLoader.java:423)
                at org.embulk.spi.Exec.doWith(Exec.java:25)
                at org.embulk.exec.BulkLoader.cleanup(BulkLoader.java:423)
                at org.embulk.EmbulkEmbed.run(EmbulkEmbed.java:184)
                ... 5 more
Caused by: java.lang.RuntimeException: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'test.test_data_table' doesn't exist
        at com.google.common.base.Throwables.propagate(Throwables.java:160)
        at org.embulk.input.jdbc.AbstractJdbcInputPlugin.transaction(AbstractJdbcInputPlugin.java:196)
        at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:544)
        ... 11 more
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'test.test_data_table' doesn't exist
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at com.mysql.jdbc.Util.handleNewInstance(Util.java:377)
        at com.mysql.jdbc.Util.getInstance(Util.java:360)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:978)
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3887)
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3823)
        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2435)
        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2582)
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2530)
        at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1907)
        at com.mysql.jdbc.PreparedStatement.execute(PreparedStatement.java:1199)
        at com.mysql.jdbc.PreparedStatement.getMetaData(PreparedStatement.java:2594)
        at org.embulk.input.jdbc.JdbcInputConnection.getSchemaOfQuery(JdbcInputConnection.java:62)
        at org.embulk.input.jdbc.AbstractJdbcInputPlugin.setupTask(AbstractJdbcInputPlugin.java:212)
        at org.embulk.input.MySQLInputPlugin.setupTask(MySQLInputPlugin.java:180)
        at org.embulk.input.jdbc.AbstractJdbcInputPlugin.transaction(AbstractJdbcInputPlugin.java:194)
        ... 12 more
Error: java.lang.RuntimeException: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'test.test_data_table' doesn't exist
root@6cd9bedd9b83:~# echo $?
0`

--- a/embulk-cli/src/main/java/org/embulk/cli/Main.java
+++ b/embulk-cli/src/main/java/org/embulk/cli/Main.java
@@ -25,6 +25,7 @@ public class Main
         }
 
         final EmbulkRun run = new EmbulkRun(EmbulkVersion.VERSION);
-        run.run(Collections.unmodifiableList(embulkArgs), Collections.unmodifiableList(jrubyOptions));
+        final int error = run.run(Collections.unmodifiableList(embulkArgs), Collections.unmodifiableList(jrubyOptions));
+        System.exit(error);
     }
 }

